examples/shopping_ads/add_shopping_smart_ad.rb

```diff
@@ -0,0 +1,274 @@
+#!/usr/bin/env ruby
+# Encoding: utf-8
+#
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This example creates a standard shopping product campaign, ad group, and ad.
+#
+# Prerequisite: You need to have access to a Merchant Center account. You can
+# find instructions to create a Merchant Center account here:
+# https://support.google.com/merchants/answer/188924
+#
+# This example creates a Smart Shopping campaign, a Smart Shopping ad group,
+# a Smart Shopping ad group ad and a listing group for "All products".
+#
+# Prerequisites
+# - You need to have access to a Merchant Center account. You can find instructions
+# to create a Merchant Center account here: https://support.google.com/merchants/answer/188924.
+# This account must be linked to your Google Ads account. The integration instructions can be
+# found at: https://developers.google.com/adwords/shopping/full-automation/articles/t15.
+# - You need your Google Ads account to track conversions. The different ways to track conversions
+# can be found here: https://support.google.com/google-ads/answer/1722054.
+
+require 'optparse'
+require 'google/ads/google_ads'
+require 'date'
+
+def add_shopping_smart_ad(
+    customer_id, merchant_center_id, create_default_listing_group)
+  # GoogleAdsClient will read a config file from
+  # ENV['HOME']/google_ads_config.rb when called without parameters
+  client = Google::Ads::GoogleAds::GoogleAdsClient.new
+
+  budget_name = add_campaign_budget(client, customer_id)
+
+  campaign_name = add_smart_shopping_campaign(
+    client,
+    customer_id,
+    budget_name,
+    merchant_center_id,
+  )
+
+  ad_group_name = add_smart_shopping_ad_group(
+    client,
+    customer_id,
+    campaign_name,
+  )
+
+  add_smart_shopping_ad_group_ad(client, customer_id, ad_group_name)
+
+  if create_default_listing_group
+    # A product group is a subset of inventory. Listing groups are the equivalent
+    # of product groups in the API and allow you to bid on the chosen group or
+    # exclude a group from bidding.
+    # This method creates an ad group criterion containing a listing group.
+    add_default_shopping_listing_group(client, customer_id, ad_group_name)
+  end
+end
+
+def add_campaign_budget(client, customer_id)
+
```

Comment:
nit: remove blank line at start of function

Reply:
done.